util/argo/resource_tracking_test.go

```diff
@@ -47,19 +47,31 @@ func TestSetAppInstanceAnnotation(t *testing.T) {
 }
 
 func TestSetAppInstanceAnnotationAndLabel(t *testing.T) {
-	yamlBytes, err := os.ReadFile("testdata/svc.yaml")
-	assert.Nil(t, err)
-	var obj unstructured.Unstructured
-	err = yaml.Unmarshal(yamlBytes, &obj)
-	assert.Nil(t, err)
+	testScenarios := map[string]string{
+		"my-app": "my-app",
+		"my-app-with-very-long-app-name-that-will-get-truncated-correctly-1234":         "my-app-with-very-long-app-name-that-will-get-truncated-correctl",
+		"my-app-with-veryvery-long-app-name-that-will-not-get-truncated-correctly-1234": "my-app-with-veryvery-long-app-name-that-will-not-get-truncated",
+		"my-app-with-a-bunch-of-hyphens-------------------------------------------1234": "my-app-with-a-bunch-of-hyphens",
```

Comment:
I wonder if, instead of cutting off early, we should just suffix the resulting name with a valid character (or sequence of characters), taken from the end of the original value.

So instead of `my-app-with-a-bunch-of-hyphens-------------------------------------------1234` becoming `my-app-with-a-bunch-of-hyphens`, it would become `my-app-with-a-bunch-of-hyphens--------------------------------4`.

Not sure if that makes sense, but it would somehow feel more expected than cutting it off very early.

Reply:
hmm!  
I think that can be a bit unexpected/confusing when it breaks and need to be debugged.  
I find it simpler and less unexpected to remove the invalid/non-alphanumeric characters post truncation.  

In this case `my-app-with-very-long-app-name-that-will-get-truncated-correctly-1234` would become `my-app-with-veryvery-long-app-name-that-will-not-get-truncated4` which is a bit weird

Reply:
Indeed, the case you mention above would be unexpected.

I meant this only for names that would truncate incorrectly, i.e. would end with an invalid character.

Reply:
Whoops; I may have done a bad copypaste job there, sorry

`my-app-with-veryvery-long-app-name-that-will-not-get-truncated-correctly-1234` will currently get incorrectly truncated to:  
`my-app-with-veryvery-long-app-name-that-will-not-get-truncated-`

With the current PR, it will be:  
`my-app-with-veryvery-long-app-name-that-will-not-get-truncated`

With your suggestion, it would be:  
`my-app-with-veryvery-long-app-name-that-will-not-get-truncated4`